server/svix-server/src/core/idempotency.rs

```diff
@@ -0,0 +1,426 @@
+// SPDX-FileCopyrightText: © 2022 Svix Authors
```

Comment:
Also actually use it, and add an integration test that checks it works (by e.g. creating an application multiple times with the same key).